docs/code/MlNetCookBook.md

```diff
@@ -244,6 +245,35 @@ var reader = mlContext.Data.TextReader(new[] {
 var data = reader.Read(dataPath);
 ```
 
+## How do I debug my experiment?
+
+Most ML.NET operations are 'lazy': they are not actually processing data, they just validate that the operation is possible, and then defer execution until the output data is actually requested. This provides good efficiency, but makes it hard to step through and debug the experiment.
+
+In order to improve debug-ability, we have added a `Preview()` extension method to all data views, transformers, estimators and readers:
+
+- `Preview` of a data view contains first 100 rows (configurable) of the data view, encoded as objects, in a single in-memory structure.
```

Comment:
>100 row [](start = 42, length = 7)

unrelated to this PR, but might be useful for users to make this configurable.

Reply:
Which it is

---
In reply to: [233184384](https://github.com/dotnet/machinelearning/pull/1612#discussion_r233184384) [](ancestors = 233184384)